packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
+/// is horizontal, and the cross axis of a [Row] is vertical.
+///
+/// [Flexible] : A widget that _could_ cause its child to stretch in the
+/// direction of the main axis, _though it does necessarily have to._
+///
+/// [Expanded] : A subtype of [Flexible] that _must_ stretch its child
+/// to use all of the space _reserved for it_ during layout.
+/// An [Expanded] will always use all of the space it is offered but this
+/// does not guarantee it will use all of the extra space in the [Flex]
+/// because it may not be offered all of the extra space within the [Flex].
+///
+/// One important thing to understand is that a [Flexible] might not use all
+/// of the space offered to it, and if it does not use all of the space
+/// offered to it then no other [Widget] will get to the rest of that space.
+/// _It is left blank._ There is an edge case in which a [Flexible] and an
+/// [Expanded] are used within the same [Flex], and the results can only be
+/// understood by comprehending what is happening "under the hood".
+/// More on this later.
+///
+/// [Flexible.fit] : The fit is the parameter of the [Flexible] that
+/// determines if it must stretch its children.
+///
+/// [FlexFit] : An enum that is used by the [Flexible.fit]. There are
+/// two values, [FlexFit.tight] and [FlexFit.loose]. These determine
+/// whether the constraints passed to the [Flexible]'s child will be
+/// tight or loose.
+///
+/// [FlexFit.tight] will force the child to stretch and use all of
+/// the space offered to it _in the main axis_. An [Expanded] is
+/// nothing more than a [Flexible] with its fit parameter hard
+/// coded to [FlexFit.tight]. It is important to remember that
+/// any size specified in the child of an [Expanded] will be ignored,
+/// as the child will be forced to be the size given by its tight
+/// constraints.
+///
+/// [FlexFit.loose] is the default fit for a [Flexible]. It will allow
+/// the child to be the size the child's size parameters state it
+/// should be, as long as that size is not outside the range defined by
+/// the minimum and maximum constraints given to the [Flexible].
+///
+/// [MainAxisSize] : An enum that determines if the [Flex] will be
+/// forced to be as large as its parent will allow.
+///
+/// [MainAxisSize.max] will force the [Flex] to be as large as the parent
+/// will allow in the main axis. _This does not force the children of the
+/// [Flex] to be larger._ This is the default.
+///
+/// [MainAxisSize.min] : Allows the [Flex] to be whatever size it needs
+/// to be in order to accommodate its children, without forcing the
+/// [Flex] to be any larger than the total size of all its children.
+///
+/// [MainAxisAlignment] : How to align, center, or apply spacing
+/// between the children in the direction of the main axis.
+///
+/// [CrossAxisAlignment] : How to align, or stretch, the children in
+/// the direction of the cross axis.
+///
+/// The multi-child layout algorithm proceeds in six steps:
```

Comment:
Maybe add a markdown title here? And maybe one for the glossary above?

Reply:
What do you mean by the glossary? Put a definition of the algorithm in the list of terms?